sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java

```diff
@@ -67,6 +72,43 @@ public ProxyOptions(ProxyAuthenticationType authentication, Proxy proxyAddress,
         }
     }
 
+    /**
+     * Attempts to load a proxy from the configuration or {@code null} if no proxy was found in the configuration.
+     *
+     * @param configuration The {@link Configuration} that is used to load proxy configurations from the environment. If
+     * {@code null} is passed then {@link Configuration#getGlobalConfiguration()} will be used. If {@link
+     * Configuration#NONE} is passed {@link IllegalArgumentException} will be thrown.
+     * @return A {@link ProxyOptions} reflecting a proxy loaded from the environment, if no proxy is found {@code null}
+     * will be returned.
+     * @throws IllegalArgumentException If {@code configuration} is {@link Configuration#NONE} or configuration is invalid.
+     */
+    public static ProxyOptions fromConfiguration(Configuration configuration) {
+        if (configuration == Configuration.NONE) {
+            throw LOGGER.logExceptionAsWarning(new IllegalArgumentException("'configuration' cannot be 'Configuration.NONE'."));
+        }
+
+        Configuration proxyConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration()
+            : configuration;
+
+        String host = proxyConfiguration.get(Properties.PROXY_HOST);
+
+        // No proxy configuration setup.
+        if (CoreUtils.isNullOrEmpty(host)) {
+            return null;
+        }
+
+        int port = proxyConfiguration.get(Properties.PROXY_PORT);
+        InetSocketAddress socketAddress = new InetSocketAddress(host, port);
```

Comment:
Does the AMQP proxy need to support creating an unresolved proxy (where the proxy isn't DNS resolved)

Reply:
good point, I also thought about it: we currently always create resolved `InetSocketAddress` in azure-core-amqp. And for HTTP options, I believe, we de-facto create unresolved one only for tests.  I decided to not change the behavior and add new features. 
If we decide to add the flag on AMQP, then I think we should add a configuration option (for both, HTTP and AMQP).